includes/AMP/Canonical_Sanitizer.php

```diff
@@ -61,6 +61,16 @@ public function sanitize() {
 
 		$query = $this->dom->xpath->query( '//link[@rel="canonical"]', $this->dom->head );
 
+		// Remove any duplicate items first.
+		if ( $query instanceof DOMNodeList && $query->length > 1 ) {
```

Comment:
There seems to two checks for `$query instanceof DOMNodeList` now. Could we something like this.

```
$rel_canonical = null;
if ( $query instanceof DOMNodeList ) {
   $rel_canonical = $query->item( 0 );
   if ( $query->length > 1 ) {
```

Reply:
Nesting the `if` statements like this would decrease readability IMHO. And `instanceof` checks are cheap, so doesn't seem worth writing it like this.